app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,399 @@
+    },
+    separator: {
+        height: 15,
+    },
+    footer: {
+        height: 40,
+        width: '100%',
+    },
+    spinner: {
+        position: 'absolute',
+        width: '100%',
+        height: '100%',
+        alignItems: 'center',
+        justifyContent: 'center',
+        zIndex: 1000,
+    },
+});
+
+const SPINNER_LAYER = 'Shape Layer 1';
+
+const EditProfile = ({
+    closeButtonId,
+    componentId,
+    currentUser,
+    isModal,
+    isTablet,
+    lockedFirstName,
+    lockedLastName,
+    lockedNickname,
+    lockedPosition,
+}: EditProfileProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const keyboardAwareRef = useRef<KeyboardAwareScrollView>();
+
+    const firstNameRef = useRef<FloatingTextInputRef>(null);
+    const lastNameRef = useRef<FloatingTextInputRef>(null);
+    const usernameRef = useRef<FloatingTextInputRef>(null);
+    const emailRef = useRef<FloatingTextInputRef>(null);
+    const nicknameRef = useRef<FloatingTextInputRef>(null);
+    const positionRef = useRef<FloatingTextInputRef>(null);
+
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = isTablet ? null : {
+        id: 'update-profile',
+        enabled: false,
+        showAsAction: 'always' as const,
+        testID: 'edit_profile.save.button',
+        color: theme.sidebarHeaderTextColor,
+        text: buttonText,
+    };
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo, isTablet]);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton!]});
+        }
+    }, [isTablet]);
```

Comment:
If we change from `!isTablet` to `isTablet`, should we remove the buttons?

Reply:
`isTablet()` (`app/utils/helpers.ts:128`) checks for the `Device.IS_TABLET` constant, and also the `isRunningInSplitView()`. I guess that running in split view does not reload the app (and therefore does not update the prop), but not sure if the hook is taking this in mind either.

Reply:
Th thing is that `isTablet` here is not using the hook, it is a prop that the component is receiving

Reply:
If this is a screen, I guess the prop will never change (it has no "parent responsible to update the prop"). So... if we foresee that the "isTablet" value may change (that I am not 100% sure, but from my previous comment I guess it could), should we use the hook instead of the prop here?

Reply:
@larkox this is a screen yes but it is also a component.

On the account screen you have different options, that on a tablet layout they will render on the right side of the device just as another componet, and on a phone it will open as a modal screen.

When the Tablet enters split mode, the right side of the layout won't be visible (in fact unmounted) and to the user will only see the `Account screen` thus will need to re-open the edit profile as a modal. That is why in this scenario there is no need to use the hook or account for the value to change

Reply:
I tested it in splitView and saw that it reloads it.  We have [isTablet](https://github.com/mattermost/mattermost-mobile/blob/4f51b1e6cbf1ac3cfd4605992555ef127ff4f015/app/screens/home/account/index.tsx#L73) constant from `home/index.tsx` that renders the `AccountTabletView` component.  This `isTablet` constant is derived from the `useIsTablet` hook

https://user-images.githubusercontent.com/43164508/145371208-e9da8677-cd35-499c-ae29-27e5c09f6923.mov

.

Reply:
That said, I do believe the dep on `isTablet` is not needed at all so it should be removed.

Reply:
I removed it. Thanks both for the in depth review.